driver/driver.cpp

```diff
@@ -54,6 +54,11 @@ Routine Description:
     WDFDEVICE hDevice = WdfWdmDeviceGetWdfDeviceHandle(DeviceObject);
     PDEVICE_EXTENSION devExt = FilterGetData(hDevice);
 
+    bool devMatch = true;
+    if (wcsncmp(L"", global.args.device_hw_id, sizeof(global.args.device_hw_id)) != 0) {
+        devMatch = wcsncmp(devExt->hwid, global.args.device_hw_id, sizeof(global.args.device_hw_id)) == 0;
```

Comment:
last arg to `wcsncmp` should be char count not size

Reply:
sizeof array gets its count

Reply:
that would be 512 * 2, when there is only 512 chars (wchars), right?

Reply:
1) according to reference: `wcsncmp` compares up to num characters
2)
```
T arr[N];
sizeof(arr) == N
```

Reply:
Well, I wasn't right.
this is ok:
```
#include <iostream>
#include <cstdint>

int main() 
{
	using T = wchar_t;
	constexpr size_t N = 512;
	
	T arr[N];
	
	std::cout << sizeof(arr) / sizeof(arr[0]) << std::endl;
}
```

Reply:
fixed.